### PR TITLE
Construct `PathBuf` from asset path before pushing asset location.

### PR DESCRIPTION
**Pull request by azriel91**
This tolerates forward slashed asset paths when the asset location was
constructed with a path containing backslashes (i.e. canonicalized
Windows paths).

Issue #622

Checklist:

* [x] Actually fix the issue + write tests for it.
* [x] cargo build --all-targets
* [x] cargo test --all
* [x] cargo run --example pong

By the way, the `.pre-commit.sh` script doesn't exist, I suppose it must've been removed at some point.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/623)

### CODE REVIEW

**Comment by azriel91** [comment]
Hold on, wrapping the call with `PathBuf::from(..)` didn't fix it. Shall find something that does

**Comment by azriel91** [comment]
Not sure if you like that fix, it's not _that_ elegant, but I couldn't think of a better way.

File: amethyst_assets/src/source/dir.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ impl Directory {
 
     fn path(&self, s_path: &str) -> PathBuf {
         let mut path = self.loc.clone();
-        path.push(s_path);
+        path.push(Path::new(s_path).iter().collect::<PathBuf>());
```
Comment (Rhuagh) [inline amethyst_assets/src/source/dir.rs]: Doesn't `PathBuf::from(s_path)` work?

Comment (Rhuagh) [reply amethyst_assets/src/source/dir.rs]: Ah, you already answered that.

Comment (torkleyy) [reply amethyst_assets/src/source/dir.rs]: Can we do the following?

```rust
path.extend(Path::new(s_path).iter());
```

That would save us some allocations.

**Comment by torkleyy** [comment]
@azriel91 Memory allocation can happen multiple times actually. For one, `collect` will push element for element and the size isn't known beforehand. So multiple allocations for `collect` + potentially one for the `push`. By using the source I posted above we don't allocate any extra buffers.

**Comment by azriel91** [comment]
Woosh, the more you know. Pushed =D

---







*Comments from [Reviewable](https://beta.reviewable.io/reviews/amethyst/amethyst/623#-:-L9P5S6L50CLR66Q3tvs:b4x0p8v)*

**Comment by torkleyy** [comment]
Okay, here you go.

bors r+

**Comment by jojolepro** [comment]
I'm late, but there's a missing new line https://github.com/amethyst/amethyst/pull/623/files#diff-6f5ca58e4d4ca434ece88e3f7cfedac1

**Pull request merged**